NitroxModel/Helper/KeyValueStore.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Runtime.InteropServices;
+using NitroxModel.Platforms.OS.Shared;
+using NitroxModel.Platforms.OS.Windows;
+
+namespace NitroxModel.Helper;
+
+/// <summary>
+///     Simple Key-Value store that works cross-platform. <br />
+/// </summary>
+/// <remarks>
+///     <para>
+///         On <b>Windows</b>:<br />
+///         Backend is <see cref="RegistryKeyValueStore" />, which uses the registry.
+///         If you want to view/edit the KeyStore, open regedit and navigate to HKEY_CURRENT_USER\SOFTWARE\Nitrox\(keyname)
+///     </para>
+///     <para>
+///         On <b>Linux</b>:<br />
+///         Backend is <see cref="ConfigFileKeyValueStore" />, which uses a file.
+///         If you want to view/edit the KeyStore, open $HOME/.config/Nitrox/nitrox.cfg in your favourite text editor.
+///     </para>
+/// </remarks>
+public class KeyValueStore : IKeyValueStore
+{
+    private static IKeyValueStore instance;
+    public static IKeyValueStore Instance { get; } = GetKeyValueStoreForPlatform();
+
+    private KeyValueStore()
+    {
+    }
+
+    public T GetValue<T>(string key, T defaultValue = default) => throw new NotSupportedException();
+
+    public bool SetValue<T>(string key, T value) => throw new NotSupportedException();
+
+    public bool DeleteKey(string key) => throw new NotSupportedException();
+
+    public bool KeyExists(string key) => throw new NotSupportedException();
+
+    private static IKeyValueStore GetKeyValueStoreForPlatform()
+    {
+        if (instance != null) 
+        {
+            return instance;
+        }
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            // Use registry on Windows
+            instance = new RegistryKeyValueStore();
+        }
+
+        // if platform isn't Windows, it doesn't have a registry
+        // use a config file for storage this should work on most platforms
+        instance = new ConfigFileKeyValueStore();
+        return instance;
```

Comment:
Accessing local state (`instance` variable) in `GetKeyValueStoreForPlatform` isn't necessary. 
`public static IKeyValueStore Instance { get; } = GetKeyValueStoreForPlatform();` means that the instance will be set when a new instance of `KeyValueStore` is created (constructor is called).

Reply:
(unfortunately, mono runtime used by Unity does not support C# 10 `static abstract members` on interfaces which would have cleaned up this code)

Reply:
I committed the changes

Reply:
Yep, it was a different bug.